am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -93,33 +94,36 @@ public void revokeResourceAccess(ExplicitAccessMetadata explicitAccessMetadata)
     }
 
     /**
-     * Returns {@link FilterResourceResponse} when record with userId and resourceId exist and has READ permissions,
-     * otherwise null.
+     * Filters {@link JsonNode} to remove fields that user has no access to (no READ permission). In addition to that
+     * method also returns map of all permissions that user has to resource.
      *
-     * @param userId       (accessorId)
-     * @param resourceId   resource id
-     * @param resourceJson json
-     * @return resourceJson or null
+     * @param userId       accessor ID
+     * @param resourceId   resource ID
+     * @param resourceJson JSON resource
+     * @return envelope {@link FilterResourceResponse} with resource ID, filtered JSON and map of permissions if access
+     *     to resource is configured, otherwise null.
      */
     public FilterResourceResponse filterResource(String userId, String resourceId, JsonNode resourceJson) {
-        ExplicitAccessRecord explicitAccess = jdbi.withExtension(AccessManagementRepository.class,
+        List<ExplicitAccessRecord> explicitAccess = jdbi.withExtension(AccessManagementRepository.class,
             dao -> dao.getExplicitAccess(userId, resourceId));
 
-        if (explicitAccess == null) {
+        if (explicitAccess.isEmpty()) {
             return null;
         }
 
-        if (READ.isGranted(explicitAccess.getPermissions())) {
-            Map<JsonPointer, Set<Permission>> attributePermissions = new ConcurrentHashMap<>();
-            attributePermissions.put(JsonPointer.valueOf(""), Permissions.fromSumOf(explicitAccess.getPermissions()));
+        Map<JsonPointer, Set<Permission>> attributePermissions = explicitAccess.stream().collect(
+            Collectors.toMap(
+                explicitAccessRecord -> JsonPointer.valueOf(explicitAccessRecord.getAttribute()),
```

Comment:
could we write getters that will return JsonPointer and Permissions.fromSumOf in the ExplicitAccessRecord model?

Reply:
Sounds like a good idea. Saw it in other place, make sense to make it consistent.

Reply:
Here is that change: https://github.com/hmcts/am-lib/pull/68/commits/a2a1973bf4152bd1d1a11b8eb80ef67169753c23